packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts

```diff
@@ -0,0 +1,371 @@
+          {
+            key,
+            name: "Delete",
+            variant: "danger",
+          },
+        ];
+
+        const detailsComponent = html`<ecc-utils-design-details
+          class="details"
+          id=${key}
+          .data=${runData}
+          .fields=${this.fields}
+          .buttons=${button}
+        >
+          <svg
+            slot="icon-${key}"
+            xmlns="http://www.w3.org/2000/svg"
+            width="16"
+            height="16"
+            fill="currentColor"
+            class="bi bi-trash3"
+            viewBox="0 0 16 16"
+          >
+            <path
+              d="M6.5 1h3a.5.5 0 0 1 .5.5v1H6v-1a.5.5 0 0 1 .5-.5M11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3A1.5 1.5 0 0 0 5 1.5v1H2.506a.58.58 0 0 0-.01 0H1.5a.5.5 0 0 0 0 1h.538l.853 10.66A2 2 0 0 0 4.885 16h6.23a2 2 0 0 0 1.994-1.84l.853-10.66h.538a.5.5 0 0 0 0-1h-.995a.59.59 0 0 0-.01 0zm1.958 1-.846 10.58a1 1 0 0 1-.997.92h-6.23a1 1 0 0 1-.997-.92L3.042 3.5zm-7.487 1a.5.5 0 0 1 .528.47l.5 8.5a.5.5 0 0 1-.998.06L5 5.03a.5.5 0 0 1 .47-.53Zm5.058 0a.5.5 0 0 1 .47.53l-.5 8.5a.5.5 0 1 1-.998-.06l.5-8.5a.5.5 0 0 1 .528-.47ZM8 4.5a.5.5 0 0 1 .5.5v8.5a.5.5 0 0 1-1 0V5a.5.5 0 0 1 .5-.5"
+            />
+          </svg>
+        </ecc-utils-design-details>`;
+
+        // Render the details component using Lit's render function
+        // Append the child to the target
+        render(detailsComponent, child);
+        target.appendChild(child);
+
+        // Add button event
+        const detailsElement = child.querySelector(
+          "ecc-utils-design-details"
+        ) as any;
+        if (detailsElement) {
+          detailsElement.addEventListener(
+            `ecc-utils-button-click`,
+            async (buttonEvent: CustomEvent) => {
+              const { key: buttonKey } = buttonEvent.detail;
+              try {
+                if (buttonKey === key) {
+                  // 0 is the index of the button as there is only one button
+                  detailsElement.setButtonLoading(0, true);
+                  const resp = (await deleteTask(this.baseURL, key)) as any;
+                  detailsElement.setButtonLoading(0, false);
+
+                  // If the response doesn't have run ID that means the run wasn't canceled.
+                  if (!resp.id) throw new Error();
+                }
+              } catch (error) {
+                eccUtilsDesignCollection.error(`Failed to cancel run: ${key}`);
+              }
+            }
+          );
+        } else {
+          eccUtilsDesignCollection.error(
+            `Failed to find ecc-utils-design-details element for run: ${key}`
+          );
+        }
+      } catch (error) {
+        eccUtilsDesignCollection.error(`Failed to fetch data for run: ${key}`);
+      }
+    }
+  }
+
+  /* TODO: Add @ecc-utils-filter when backend implements filtering */
+  render() {
+    return html`
+      <ecc-utils-design-collection
+        id="collection"
+        .filters=${this.filters}
```

Comment:
Remove filter we will handle them later!

Reply:
done